lib/appium-driver.d.ts

```diff
@@ -164,18 +162,26 @@ export declare class AppiumDriver {
      * @param inertia
      * @param xOffset
      */
-    swipe(y: number, x: number, yOffset: number, inertia?: number, xOffset?: number): Promise<void>;
+    swipe(startPoint: {
+        y: number;
+        x: number;
+    }, endPoint: {
+        y: number;
+        x: number;
+    }, inertia?: number): Promise<void>;
     /**
     * Click a point by providing coordinates
     * @param x
     * @param y
     */
     clickPoint(xCoordinate: number, yCoordinate: number): Promise<void>;
+    getOrientation(): Promise<DeviceOrientaion>;
+    setOrientation(orientation: DeviceOrientaion): Promise<void>;
     source(): Promise<any>;
     sessionId(): Promise<any>;
-    compareElement(element: UIElement, imageName: string, tolerance?: number, timeOutSeconds?: number, toleranceType?: ImageOptions): Promise<boolean>;
-    compareRectangle(rect: IRectangle, imageName: string, timeOutSeconds?: number, tolerance?: number, toleranceType?: ImageOptions): Promise<boolean>;
-    compareScreen(imageName: string, timeOutSeconds?: number, tolerance?: number, toleranceType?: ImageOptions): Promise<boolean>;
+    compareElement(element: UIElement, imageName?: string, tolerance?: number, timeOutSeconds?: number, toleranceType?: ImageOptions): Promise<boolean>;
```

Comment:
Why imageName is optional? If we don't provide image, what would be compared to the element?

Reply:
This is a good question and I have missed to mention. Generally, the image name could be provided by imageHelper.testName or pass as an argument. It is up to you. 
This is very convenience when you use something like this 

`    beforeEach(async function () {
        driver.imageHelper.testName = this.currentTest.title;
    });
`
And more over in case you need to make a few image comparison in one test the image helper will automatically add an index to each next image.